src/yaml-tools.js

```diff
@@ -111,11 +123,30 @@ yamlLibrary = {
   // doc: The SAM.YML parsed document
   // schema: the JSON schema to compare the doc to
   checkDocAgainstSchema: function (doc, schema, failed, warning, info) {
-    this.doc = doc;
+    this.doc = this.docs[doc];
     var keys = [];
 
     this.recurseIntoArray(schema, keys, null, failed, warning, info);
+  },
+
+  loadDocData: function (docs) {
+    let docsArray = cleanDocsData(docs).split('---');
+
+    for (let i = 0; i < docsArray.length; i++) {
+      let allComments = true;
+      let yamlDoc = yaml.load(docsArray[i]);
+      this.docs[i] = yamlDoc == null || yamlDoc == undefined ? {} : yamlDoc;
+
+      let lines = docsArray[i].split('\n');
+      for (let j = 0; j < lines.length; j++) {
+        let line = lines[j].trim();
+        if (line.length > 0 && !line.startsWith('#')) {
```

Comment:
I don't think this situation is going to come up since it seems like cleanDocsData is removing the comments

Reply:
👍🏽